pkgs/development/misc/msp430/mspds/boost.patch

```diff
@@ -0,0 +1,131 @@
+diff --git a/DLL430_v3/src/TI/DLL430/UsbCdcIoChannel.cpp b/DLL430_v3/src/TI/DLL430/UsbCdcIoChannel.cpp
+index 78be8cd..03111fb 100644
+--- a/DLL430_v3/src/TI/DLL430/UsbCdcIoChannel.cpp
++++ b/DLL430_v3/src/TI/DLL430/UsbCdcIoChannel.cpp
+@@ -42,7 +42,7 @@
+ 
+ #include <boost/asio/read.hpp>
+ #include <boost/asio/write.hpp>
+-#include <boost/asio/io_service.hpp>
++#include <boost/asio/io_context.hpp>
+ 
+ #if defined(_WIN32) || defined(_WIN64)
+ 
+@@ -89,7 +89,7 @@ using namespace boost::asio;
+ UsbCdcIoChannel::UsbCdcIoChannel(const PortInfo& portInfo)
+  : UsbIoChannel(portInfo)
+  , inputBuffer(260)
+- , ioService(0)
++ , ioContext(0)
+  , port(0)
+  , comState(ComStateRcv)
+  , bytesReceived(0)
+@@ -376,9 +376,9 @@ std::string UsbCdcIoChannel::retrieveSerialFromId(const std::string& id)
+ 
+ bool UsbCdcIoChannel::openPort()
+ {
+-	ioService = new boost::asio::io_service;
+-	port = new boost::asio::serial_port(*ioService);
+-	timer = new boost::asio::deadline_timer(*ioService);
++	ioContext = new boost::asio::io_context;
++	port = new boost::asio::serial_port(*ioContext);
++	timer = new boost::asio::deadline_timer(*ioContext);
+ 
+ 	boost::system::error_code ec;
+ 	ec = port->open(portInfo.path, ec);
+@@ -466,8 +466,8 @@ void UsbCdcIoChannel::cleanup()
+ 	timer = 0;
+ 	delete port;
+ 	port = 0;
+-	delete ioService;
+-	ioService = 0;
++	delete ioContext;
++	ioContext = 0;
+ }
+ 
+ bool UsbCdcIoChannel::close()
+@@ -542,7 +542,7 @@ size_t UsbCdcIoChannel::read(HalResponse& resp)
+ 
+ 	boost::system::error_code ec;
+ 
+-	while (ioService->run_one(ec))
++	while (ioContext->run_one())

```

Comment:
Per boost documentation (https://www.boost.org/doc/libs/1_70_0/doc/html/boost_asio/reference/io_context/run.html) the non-error_code overload should be preferred